server/src/main/java/org/candlepin/policy/js/autobind/AutobindRules.java

```diff
@@ -98,6 +98,7 @@ public AutobindRules(JsRunner jsRules, ProductCache productCache) {
         args.put("compliance", compliance);
         args.put("exemptList", exemptLevels);
         args.put("considerDerived", considerDerived);
+        args.put("newborn", consumer.isNewborn());
```

Comment:
Not a big fan of seeing newborn passed into rules explicitly in ~5 places.

Looking at how it's used in the rules file, I'd wager that will break on an old server where newborn isn't passed in. The usage in JS needs to be updated to safely handle if it's not there. (doesn't look like it's ok as is, but I might be wrong)

Larger issue though is whether we should be passing a piece of data that small, in 5 places. I can see it was done so we don't have to calculate it in two palces and two languages though.

For one it's just a tiny piece of data and should be relatively easy to calculate in JS. Passing data to rules needs to be done carefully, as an example I think this would break rules on old servers who would expect newborn to be set in the context. (I don't think the usage below looks safe) We can adjust that but I'm still not sure it should be passed explicitly. Feels more like something the rules should figure out themselves.

Looking at how it's used in the rules, I'm not sure we're safely guarding against it not being there if we ship these new rules down to an old server. We need to be sure we're handling the case that it's not set.